engine/src/main/scala/cromwell/engine/workflow/SingleWorkflowRunnerActor.scala

```diff
@@ -84,32 +109,35 @@ case class OldStyleSingleWorkflowRunnerActor(source: WorkflowSourceFiles,
   when (RunningWorkflow) {
     case Event(WorkflowManagerSubmitSuccess(id), data) =>
       log.info(s"$tag: Workflow submitted UUID($id)")
-      workflowManager ! SubscribeToWorkflowCommand(id)
+      schedulePollRequest()
       stay() using data.copy(id = Option(id))
-    case Event(Transition(_, _, WorkflowSucceededState), data) =>
-      workflowManager ! WorkflowOutputs(data.id.get)
+    case Event(IssuePollRequest, _) =>
+      requestStatus()
+      stay()
+    case Event(RequestComplete((StatusCodes.OK, jsObject: JsObject)), data) if !jsObject.state.isTerminal =>
```

Comment:
Yes I think you're right.  The generic `whenUnhandled` case doesn't really capture the essence of "not okay" in `StatusCode != OK`.  😄